src/lightning/__setup__.py

```diff
@@ -24,6 +25,39 @@ def _load_py_module(name: str, location: str) -> ModuleType:
 _SETUP_TOOLS = _load_py_module("setup_tools", os.path.join(_PROJECT_ROOT, ".actions", "setup_tools.py"))
 
 
+def _prepare_extras() -> Dict[str, Any]:
+    # https://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-extras
+    # Define package extras. These are only installed if you specify them.
+    # From remote, use like `pip install pytorch-lightning[dev, docs]`
+    # From local copy of repo, use like `pip install ".[dev, docs]"`
+    req_files = glob.glob(os.path.join(_PATH_REQUIREMENTS, "*", "*.txt"))
+    reqs = [dict(dir=os.path.dirname(p), fname=os.path.basename(p)) for p in req_files]
```

Comment:
The code would be simpler to read if you convert everything to Paths

```suggestion
    reqs = [Path(p) for p in req_files]
```

and then access `r.{name,dirname,stem}` etc below. See the docs: https://docs.python.org/3/library/pathlib.html#correspondence-to-tools-in-the-os-module

Reply:
updated